python/pyspark/sql/dataframe.py

```diff
@@ -4430,6 +4430,48 @@ def withColumnRenamed(self, existing: str, new: str) -> "DataFrame":
         """
         return DataFrame(self._jdf.withColumnRenamed(existing, new), self.sparkSession)
 
+    def withColumnsRenamed(self, *colsMap: Dict[str, str]) -> "DataFrame":
+        """
+        Returns a new :class:`DataFrame` by renaming multiple columns.
+        This is a no-op if schema doesn't contain the given column names.
+
+        The colsMap is a map of existing column name and corresponding desired column names.
```

Comment:
Maybe we don't need to describe the parameter detail in the summary, since it's duplicated information with `Parameters` section ??